### PR TITLE
add support for global dynamic log level

### PR DESCRIPTION
**Pull request by greg-mclain**
**Description**
Adds the ability to change the global log level of an application.

### CODE REVIEW

File: log/log.go [hunk 1]
```diff
@@ -147,3 +151,9 @@ func Get(ctx context.Context) *zap.Logger {
 	}
 	return logger
 }
+
+// RegisterLogLevelHandler registers an endpoint handler with the specified router to change the global log level
+// for the application.
+func RegisterLogLevelHandler(router *mux.Router) {
+	router.Handle("/loglevel", globalLogLevel)
```
Comment (theisenmark) [inline log/log.go:158]: This is super cool! Can we add a bit more documentation around this? Maybe here or in the README or something? I had to dive into the [zap docs](https://github.com/uber-go/zap/blob/master/http_handler.go#L34-L36) to figure out exactly how to use this endpoint.

**Review by theisenmark — APPROVED** [review]

**Review by dj80hd — APPROVED** [review]
Very cool

File: log/log.go (hunk 1 above)
Comment (greg-mclain) [reply log/log.go:159]: Good suggestion. I added it in the function documentation for now, although this is a reminder that we don't have great documentation in general on how to use several of the tools here. We should circle back and either fill out the root level README better and/or add READMEs for each of the components.

**Pull request merged**